src/main/java/com/champions/carsharingservice/service/impl/UserServiceImpl.java

```diff
@@ -27,10 +29,35 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         if (userRepository.existsByEmail(requestDto.email())) {
             throw new RegistrationException("Unable to complete registration. User already exists");
         }
-        Role defaultRole = roleRepository.findRoleByName(Role.RoleName.CUSTOMER);
+        Role defaultRole = roleRepository.findRoleByName(Role.RoleName.CUSTOMER).orElseThrow();
         User user = userMapper.toUser(requestDto);
         user.setRoles(Set.of(defaultRole));
         user.setPassword(passwordEncoder.encode(requestDto.password()));
         return userMapper.toUserResponse(userRepository.save(user));
     }
+
+    @Override
+    public UserResponseDto updateUserRole(Long id, String newRole) {
```

Comment:
i think its better to make it transactional and it will be not neccesarily to save user